test/UtilityFunctionsTests.cpp

```diff
@@ -69,4 +69,18 @@ SCENARIO("check sumNumbersFromGenerators() function")
             }
         }
     }
+    GIVEN("Invalid generators collection")
+    {
+        std::vector<GeneratorType> collection = { static_cast<GeneratorType>(-1) };
+        auto numberOfValues = 1;
+
+        WHEN("sumNumbersFromGenerators is called")
+        {
+            auto result = sumNumbersFromGenerators(collection, numberOfValues);
+            THEN("It should return -1")
```

Comment:
For me, expected output should be an exception.
-1 is a valid number, which can be in some cases returned from generators.

Reply:
Yep, with next generator negative value could be correct (for prime & Fibonacci numbers negative values are not correct).